File: 2023/day_1/elixir/part_one.exs
{:ok, content} = File.read("../input.txt")

nums = Map.new(0..9, fn num -> {to_string(num), num} end)

content
|> String.split("\n")
|> Enum.map(&String.graphemes/1)
|> Enum.map(&Enum.filter(&1, fn grapheme -> Enum.member?(Map.keys(nums), grapheme) end))
|> Enum.map(&[nums[List.first(&1)] || 0, nums[List.last(&1)] || 0])
|> Enum.map(fn [first, last] -> 10 * first + last end)
|> Enum.sum()
|> IO.inspect(label: "Answer")

File: 2024/day_1/js/part_one.js
import { readFileSync } from "node:fs";

const input = readFileSync("input.txt", "utf-8");

let nums = input.split("\n").map((x) =>
  x
    .split(" ")
    .filter((x) => x !== "")
    .map((x) => parseInt(x, 10))
);

let left = nums.map((x) => x[0]).sort((a, b) => a - b);
let right = nums.map((x) => x[1]).sort((a, b) => a - b);

let sum = 0;

for (let i = 0; i < left.length; i++) {
  sum += Math.abs(left[i] - right[i]);
}

console.log(sum);
